src/test/java/net/obvj/performetrics/util/DurationUtilsTest.java

```diff
@@ -0,0 +1,108 @@
+package net.obvj.performetrics.util;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static net.obvj.performetrics.util.DurationUtils.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.junit.Test;
+
+public class DurationUtilsTest
+{
+    private static final Duration D_ZERO = Duration.ZERO;
+    private static final Duration D_500_MILLIS = Duration.of(500, MILLISECONDS);
+    private static final Duration D_750_MILLIS = Duration.of(750, MILLISECONDS);
+    private static final Duration D_1250_MILLIS = Duration.of(1250, MILLISECONDS);
+    private static final Duration D_1_SECOND = Duration.of(1, SECONDS);
+    private static final Duration D_2_SECONDS = Duration.of(2, SECONDS);
+
+    @Test
+    public void average_null_zero()
+    {
+        assertThat(average(null), is(equalTo(D_ZERO)));
+    }
+
+    @Test
+    public void average_empty_zero()
+    {
+        assertThat(average(Collections.emptyList()), is(equalTo(D_ZERO)));
+    }
+
+    @Test
+    public void average_oneElement_sameValue()
+    {
+        assertThat(average(Collections.singleton(D_500_MILLIS)), is(equalTo(D_500_MILLIS)));
+    }
+
+    @Test
+    public void average_twoOrMoreElements_meanValue()
+    {
+        assertThat(average(Arrays.asList(D_500_MILLIS, D_1_SECOND)), is(equalTo(D_750_MILLIS)));
+        assertThat(average(Arrays.asList(D_500_MILLIS, D_2_SECONDS)), is(equalTo(D_1250_MILLIS)));
+        assertThat(average(Arrays.asList(D_500_MILLIS, D_500_MILLIS, D_2_SECONDS)), is(equalTo(D_1_SECOND)));
+    }
+
+    @Test
+    public void average_containingNullElements_nullElementsDontCount()
+    {
```

Comment:
Add a test to secure the behavior when the collection contains only null elements